### DISCUSSION TITLE
Issue with web3Auth not properly connecting to Metamask

### DISCUSSION THREAD

**Discussion by gabyteera** [Web3Auth/Web3Auth#1296] [Help]
Hey there, I've been having an issue with web3Auth and could use some help. Sometimes, when I try to connect my MetaMask, the web3Auth modal prompts me for verification, but the MetaMask window doesn't pop up. Even when I click on the extension, nothing happens - it just shows my account. This issue keeps reoccurring in several scenarios, for example:
	1.	When I first build my Next.js project and the cachedAdapter and connectedAdapterName of web3Auth are null (in this case I have reinstalled my MetaMask and the account has never been connected to the DApp previously).
	2.	When the cachedAdapter is "metamask", but the connectedAdapterName is null. In this case, the login options are not provided, web3Auth modal instantly displays "Verify on your Account to continue" without a MetaMask login image - the resource fails to load. (This happens when the account has been previously connected to the DApp, I close the window without logging out and then re-open the page in a new tab).
The issue resolves itself if I refresh the page. I would really appreciate any help you can offer to resolve it. Thanks!

**Comment by ieow** [comment]
Could you provide image of the issue and the logs ?

**Reply by gabyteera** [reply]
### Scenario 1: 

When user opens the site for the first time with disconnected Metamask account (in this case Next.js project was just built)

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/99966036/217568093-05af059d-d173-45ce-ba25-77d2c69b7f82.png)

User clicks connect button and Web3Auth login options are displayed. User chooses Metamask, Web3Auth prompts for verification, but Metamask window does not pop up

![7a9fac08-5351-4d34-90eb-d4eacac23b98](https://user-images.githubusercontent.com/99966036/217556092-3696b0c7-b0fd-447f-8ca9-74f3222e4efc.jpg)
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/99966036/217568358-cd8aa2e7-9ee0-42a3-814b-371330d8871a.png)


If the page is reloaded - everything works as expected. Again, Web3Auth asks for verification, but this time Metamask window pops up.

![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/99966036/217568947-8f6dc9d3-3a2a-43eb-ae7c-16cb0c65524e.png)



### Scenario 2: 
User has previously connected Metamask account to the site, has not logged-out. The window has been closed and re-opened in another tab. When connect button is clicked - no login options are provided, the following modal is displayed with the following logs:

![5d273a93-8272-452a-b35d-32ab9ca5a326](https://user-images.githubusercontent.com/99966036/217560855-b29e98dd-a7da-411f-b0ab-a5d028768f41.jpg)
![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/99966036/217569034-aa434efb-3b2c-4d8d-9a85-fd3323333c2e.png)


When the page is reloaded, everything works fine. When the connect button is clicked - web3Auth automatically re-connects the account to the site. 

![MicrosoftTeams-image (6)](https://user-images.githubusercontent.com/99966036/217569797-54b7a8fe-eda5-4a05-b6d0-21bf8cc4c383.png)

**Reply by BhargaviDetasecure** [reply]
@gabyteera did you solve this issue?

**Reply by pawanlapits** [reply]
Hi @gabyteera  I am also facing a similar issue. is this resolved? please let me know how you fixed that
